internal/configfsm/fsm.go

```diff
@@ -0,0 +1,309 @@
+package configfsm
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/qmuntal/stateless"
+)
+
+type Trigger string
+
+const (
+	// Transitions to Missing state
+	StartToMissing           Trigger = "Start->Missing"
+	ClusterNotReadyToMissing Trigger = "ClusterNotReady->Missing"
+	InProgressToMissing      Trigger = "InProgress->Missing"
+	CompletedToMissing       Trigger = "Completed->Missing"
+	TimedOutToMissing        Trigger = "TimedOut->Missing"
+	OutOfDateToMissing       Trigger = "OutOfDate->Missing"
+
+	// Transitions to ClusterNotReady state
+	MissingToClusterNotReady    Trigger = "Missing->ClusterNotReady"
+	InProgressToClusterNotReady Trigger = "InProgress->ClusterNotReady"
+	CompletedToClusterNotReady  Trigger = "Completed->ClusterNotReady"
+	OutOfDateToClusterNotReady  Trigger = "OutOfDate->ClusterNotReady"
+
+	// Transitions to InProgress state
+	ClusterNotReadyToInProgress Trigger = "ClusterNotReady->InProgress"
+	OutOfDateToInProgress       Trigger = "OutOfDate->InProgress"
+	CompletedToInProgress       Trigger = "Completed->InProgress"
+
+	// Transitions to TimedOut state
+	ClusterNotReadyToTimedOut Trigger = "ClusterNotReady->TimedOut"
+	InProgressToTimedOut      Trigger = "InProgress->TimedOut"
+
+	// Transitions to Completed state
+	InProgressToCompleted Trigger = "InProgress->Completed"
+
+	// Transitions to OutOfDate state
+	InProgressToOutOfDate      Trigger = "InProgress->OutOfDate"
+	TimedOutToOutOfDate        Trigger = "TimedOut->OutOfDate"
+	ClusterNotReadyToOutOfDate Trigger = "ClusterNotReady->OutOfDate"
+	CompletedToOutOfDate       Trigger = "Completed->OutOfDate"
+
+	// Self transitions
+	MissingToMissing                 Trigger = "Missing->Missing"
+	ClusterNotReadyToClusterNotReady Trigger = "ClusterNotReady->ClusterNotReady"
+	InProgressToInProgress           Trigger = "InProgress->InProgress"
+	TimedOutToTimedOut               Trigger = "TimedOut->TimedOut"
+	CompletedToCompleted             Trigger = "Completed->Completed"
+	OutOfDateToOutOfDate             Trigger = "OutOfDate->OutOfDate"
+
+	// States
```

Comment:
It would be nice here to use the Conditions from the [utils package](https://github.com/openshift-kni/oran-o2ims/blob/main/internal/controllers/utils/conditions.go#L52-L72).

Reply:
Fixed